### PR TITLE
docs: design workflow for xblock skill tagging

### PR DESCRIPTION
**Pull request by navinkarkera**
Adds document describing the whole workflow for xblock skill tagging.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @navinkarkera!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by navinkarkera** [comment]
@sameenfatima78 Please let me know your thoughts on the document as well as whether this is a good place to keep it.

File: docs/decisions/0001-xblock-skill-tagging-design.rst
Comment (robrap) [inline docs/decisions/0001-xblock-skill-tagging-design.rst:59]: Minor typo.

**Review by robrap — COMMENTED** [review]
1. This is immensely helpful. Thank you!
2. Were there any Rejected Alternatives to document?
3. I’ll leave the more detailed review to the teams involved.

**Comment by mphilbrick211** [comment]
Hi @moconnell1453! Flagging this for your review.

**Review by moconnell1453 — APPROVED** [review]

File: docs/decisions/0001-xblock-skill-tagging-design.rst
Comment (navinkarkera) [reply docs/decisions/0001-xblock-skill-tagging-design.rst]: Thanks. Fixed it.

**Comment by navinkarkera** [comment]
@robrap Thank you!

> Were there any Rejected Alternatives to document?

No.

**Comment by navinkarkera** [comment]
@moconnell1453 Thank you! I cannot merge it as I don't have merge rights.

**Review by sameenfatima78 — APPROVED** [review]
Looks awesome @navinkarkera, thanks for this!

**Comment by sameenfatima78** [comment]
> @moconnell1453 Thank you! I cannot merge it as I don't have merge rights.

@navinkarkera I'll take care of it.

**Comment by openedx-webhooks** [comment]
@navinkarkera 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/taxonomy-connector&entry.867055334=https://github.com/openedx/taxonomy-connector/pull/140&entry.1484655318=https://github.com/navinkarkera&entry.752974735=2023-02-03+10:36&entry.1917517419=2023-02-06+10:21&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
